clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -322,6 +323,25 @@ public void testOffsetsForTimesOnNullPartitions() {
                 Duration.ofMillis(1)));
     }
 
+    @Test
+    public void testOffsetsForTimesFailsOnNegativeTargetTimes() {
+        PrototypeAsyncConsumer<?, ?> consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
+        assertThrows(IllegalArgumentException.class,
+                () -> consumer.offsetsForTimes(Collections.singletonMap(new TopicPartition(
+                                "topic1", 1), ListOffsetsRequest.EARLIEST_TIMESTAMP),
+                        Duration.ofMillis(1)));
+
+        assertThrows(IllegalArgumentException.class,
+                () -> consumer.offsetsForTimes(Collections.singletonMap(new TopicPartition(
```

Comment:
Why do we test this for these three constants and not for `EARLIEST_LOCAL_TIMESTAMP`?

Reply:
Actually, under which circumstances would a user even use these constants? All uses seem to be internal, if I see correctly. Checking non-negative makes sense, just not sure why we are checking for these (internal?) constants specifically.

Reply:
These constants are defined public in the [ListOffsetsRequest](https://github.com/apache/kafka/blob/4cf86c5d2f6355c5684ecedeb1bf96e90877244d/clients/src/main/java/org/apache/kafka/common/requests/ListOffsetsRequest.java#L41-L43). 
You're right that the validation is indeed for all negative values, but this test is what I expect could be the most common misuse of this API (a user trying to retrieve offsets using an "abstract" timestamp like EARLIEST or LATEST). This also was the initial motivation for the validation in the old consumer with [this](https://github.com/apache/kafka/blob/4cf86c5d2f6355c5684ecedeb1bf96e90877244d/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L2115) comment, so aligning with it. Makes sense?

Reply:
Consistency with the old consumer, say no more!